src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -91,20 +94,54 @@ public void setupSchema(List<ColumnMetadata> columns) {
       if (!column.getNullable()) {
         addNonNullableFieldName(column.getInternalName());
       }
-      this.statsMap.put(
-          column.getInternalName(),
-          new RowBufferStats(column.getName(), column.getCollation(), column.getOrdinal()));
-
-      if (onErrorOption == OpenChannelRequest.OnErrorOption.ABORT
-          || onErrorOption == OpenChannelRequest.OnErrorOption.SKIP_BATCH) {
-        this.tempStatsMap.put(
+      if (!clientBufferParameters.getIsIcebergMode()) {
+        /* Streaming to FDN table doesn't support sub-columns, set up the stats here. */
+        this.statsMap.put(
             column.getInternalName(),
-            new RowBufferStats(column.getName(), column.getCollation(), column.getOrdinal()));
+            new RowBufferStats(
+                column.getName(), column.getCollation(), column.getOrdinal(), null /* fieldId */));
+
+        if (onErrorOption == OpenChannelRequest.OnErrorOption.ABORT
+            || onErrorOption == OpenChannelRequest.OnErrorOption.SKIP_BATCH) {
+          this.tempStatsMap.put(
+              column.getInternalName(),
+              new RowBufferStats(
+                  column.getName(),
+                  column.getCollation(),
+                  column.getOrdinal(),
+                  null /* fieldId */));
+        }
       }
 
       id++;
     }
     schema = new MessageType(PARQUET_MESSAGE_TYPE_NAME, parquetTypes);
+    if (clientBufferParameters.getIsIcebergMode()) {
+      /* Iceberg mode requires stats for sub-columns, set them up here. */
+      for (ColumnDescriptor subColumnDescriptor : schema.getColumns()) {
+        String subColumnName = concatDotPath(subColumnDescriptor.getPath());
+
+        /* set fieldId to 0 for non-structured columns */
```

Comment:
do you mind adding in a big comment block here what are the expected fieldId and ordinal values for each field from the example we had in the conf room?
INT F1,
STRUCT F2 { 
    STRUCT F21 { 
        INT F212 
    }, 
    INT F22 
},
INT F3,
MAP F4 {
    LONG F4K,
    MAP F4V {
        STR F4VK,
        STR F4VV
    }
},
INT F5,
ARRAY F6,
INT F7

Reply:
Added a comment block with an example to explain `fieldId`.